mobile/android/android-emulator-services.ts

```diff
@@ -150,17 +253,45 @@ class AndroidEmulatorServices implements Mobile.IEmulatorPlatformServices {
 				this.$errors.fail(AndroidEmulatorServices.UNABLE_TO_START_EMULATOR_MESSAGE);
 			}
 
-			var emulatorId = _.first(runningEmulators);
-			if(options.avd) {
-				// get the id of the started emulator
-				_.forEach(runningEmulators, (emulator) => {
-					if(!_.contains(initiallyRunningEmulators, emulator)) {
-						emulatorId = emulator;
-					}
-				});
+			return emulatorId;
+		}).future<string>()();
+	}
+
+	private getRunningGenymotionEmulators(adbDevicesOutput: string[]): IFuture<string[]> {
+		return ((): string[]=> {
+			var genymotionEmulators: string[] = [];
+			var futures: IFuture<string>[] = [];
+			_(adbDevicesOutput).filter(r => !r.match(AndroidEmulatorServices.RUNNING_ANDROID_EMULATOR_REGEX)).each(row => {
+				var match = row.match(/^(.+?)\s+device$/);
+				if(match && match[1]) {
+					// possible genymotion emulator
+					var emulatorId = match[1];
+					var future = this.checkForGenymotionProductManufacturer(emulatorId);
+					futures.push(future);
+				}
+			});
```

Comment:
You can do improve this: `var futures = _(adbDevicesOutput).filter(r => !r.match(AndroidEmulatorServices.RUNNING_ANDROID_EMULATOR_REGEX)).map(row => {
				var match = row.match(/^(.+?)\s+device$/);
				if(match && match[1]) {
					// possible genymotion emulator
					var emulatorId = match[1];
					return this.checkForGenymotionProductManufacturer(emulatorId);
				}`